pkgs/development/python-modules/yamlfix/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, fetchPypi
+, pathspec
+, pytest
+, pythonOlder
+, click
+, ruyaml
+, stdenv
+}:
+
+buildPythonPackage rec {
+  pname = "yamlfix";
+  version = "0.7.2";
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1fdea35a6865c3332a227e0b75ed4c31d0d1a442c0f770353bd23406511fece9";
+  };
+
+  propagatedBuildInputs = [
+    click
+    ruyaml
+  ];
+
+  checkInputs = [
+    pytest
```

Comment:
Please use pytestCheckHook.

Reply:
Running: `nix-env -f $(git rev-parse --show-toplevel) -iA yamlfix -vvv`
Fails with `pytestCheckHook` but succeeds with `pytest`. The binary works for my on `Linux 5.13.19_1 x86_64 GNU/Linux`.
Does pytestCheckHook cover more architectures or does it bring more stability?

Reply:
pytestCheckHook allows to easy disable failing tests or files and all new packages should use it. How does it fail?

Reply:
This is the error with some info before and after.

```text
Rewriting #!/nix/store/k0z9n599k02hab8qjjp3ljw065iwjcvg-python3-3.9.6/bin/python3.9 to #!/nix/store/k0z9n599k02hab8qjjp3ljw065iwjcvg-python3-3.9.6
wrapping `/nix/store/p0d8y29l2048mlhsib62ksbkm1d6c7hc-python3.9-yamlfix-0.7.2/bin/yamlfix'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
running install tests
no Makefile or custom installCheckPhase, doing nothing
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: yamlfix
pytestCheckPhase
Executing pytestCheckPhase
ERROR: usage: __main__.py [options] [file_or_dir] [file_or_dir] [...]
__main__.py: error: unrecognized arguments: -n
  inifile: /build/yamlfix-0.7.2/pyproject.toml
  rootdir: /build/yamlfix-0.7.2

building of '/nix/store/cfx2vavjv8mcghj20liys8rpidzs511w-python3.9-yamlfix-0.7.2.drv': got EOF
building of '/nix/store/cfx2vavjv8mcghj20liys8rpidzs511w-python3.9-yamlfix-0.7.2.drv': woken up
building of '/nix/store/cfx2vavjv8mcghj20liys8rpidzs511w-python3.9-yamlfix-0.7.2.drv': build done
killing process 23531
builder process for '/nix/store/cfx2vavjv8mcghj20liys8rpidzs511w-python3.9-yamlfix-0.7.2.drv' finished
killing all processes running under uid '30001'
builder for '/nix/store/cfx2vavjv8mcghj20liys8rpidzs511w-python3.9-yamlfix-0.7.2.drv' failed with exit code 4
lock released on '/nix/store/p0d8y29l2048mlhsib62ksbkm1d6c7hc-python3.9-yamlfix-0.7.2.lock'
building of '/nix/store/cfx2vavjv8mcghj20liys8rpidzs511w-python3.9-yamlfix-0.7.2.drv': done
building of '/nix/store/cfx2vavjv8mcghj20liys8rpidzs511w-python3.9-yamlfix-0.7.2.drv': goal destroyed
error: build of '/nix/store/cfx2vavjv8mcghj20liys8rpidzs511w-python3.9-yamlfix-0.7.2.drv' failed
```

I am running NixOS 21.11 now, but I've had the same error on VoidLinux before.
I'm not familiar with pytestCheckHook, I do not know what it actually tries to run.

Can I find that command somewhere?

Reply:
Looks like that `pytest-xdist` is missing in `checkInputs`. `pytestFlagsArray = [ "-n" "$NIX_BUILD_CORES" ];` should help.

Reply:
Using `pytestCheckHook`, that results in: `unrecognized arguments: -n -n 8`

```
Check whether the following modules can be imported: yamlfix
pytestCheckPhase
Executing pytestCheckPhase
ERROR: usage: __main__.py [options] [file_or_dir] [file_or_dir] [...]
__main__.py: error: unrecognized arguments: -n -n 8
  inifile: /build/yamlfix-0.7.2/pyproject.toml
  rootdir: /build/yamlfix-0.7.2
```

I will check in the non-working pytestCheckHook variant, maybe some automated check has more information.